src/sentry/api/bases/sentryapps.py

```diff
@@ -212,10 +223,15 @@ class SentryAppBaseEndpoint(IntegrationPlatformEndpoint):
     permission_classes = (SentryAppPermission,)
 
     def convert_args(self, request: Request, sentry_app_slug, *args, **kwargs):
-        try:
-            sentry_app = SentryApp.objects.get(slug=sentry_app_slug)
-        except SentryApp.DoesNotExist:
-            raise Http404
+        if SiloMode.get_current_mode() in [SiloMode.MONOLITH, SiloMode.CONTROL]:
+            try:
+                sentry_app = SentryApp.objects.get(slug=sentry_app_slug)
+            except SentryApp.DoesNotExist:
+                raise Http404
+        else:
+            sentry_app = app_service.get_sentry_app_by_slug(slug=sentry_app_slug)
+            if sentry_app is None:
+                raise Http404
```

Comment:
Should we always use the service to get the sentryapp? That would let us have more consistent typings and behavior across endpoints? I know that using both ORM access and service access lets us churn fewer lines of code but it could make future maintenance more challenging.

Reply:
We just need to make a call on this. I'm worried about the scope of the code churn, honestly. The amount of stuff we have to touch is already large and taking on more seems like it might put our timelines at risk. I'm leaning toward getting it all working, then take these out, which would give you a clear sense of scope for the undertaking.

Reply:
>  I'm worried about the scope of the code churn, honestly. The amount of stuff we have to touch is already large and taking on more seems like it might put our timelines at risk.

Thats's fair. The scope of changes could get large. Lets go with this approach then as it helps us limit the scope today and we can rework this later to always use the RpcUser in the future.